### DISCUSSION TITLE
How to add attributes to link elements?

### DISCUSSION THREAD

**Discussion by sunglee** [markdoc/markdoc#194] [Q&A] [answered]
I was wondering if it is possible to add `rel` attributes to anchor elements.
I was thinking this would work, but it's not.

```
const ast = Markdoc.parse(md);

for (const node of ast.walk()) {
  if (node.type === 'link') {
    // node.attributes.href = '/some-url'; <-- this updates href when rendered.
    // node.attributes.target = '_blank'; <-- this doesn't work.
    node.attributes.rel = 'noopener noreferrer'; <-- this doesn't work either.
    console.log({ node }); <-- the console output has all attributes in an object but rendered output only updates 'href'.
  }
}
```

How can I add/update attributes?

Thanks for your help!

**Comment by mfix-stripe** [comment, marked as answer]
@sunglee you just need to add `rel` and `target` as attributes on your `link` node configuration:

```js
// markdoc/nodes/link.markdoc.js

import { link } from '@markdoc/next.js/tags';

export default {
  ...link,
  render: 'a',
  attributes: {
    ...link.attributes,
    rel: {
      type: String
    },
    target: {
      type: String
    }
  },
};
```

I tested this out locally and it works:

<img width="1512" alt="Screen Shot 2022-09-12 at 5 17 34 PM" src="https://user-images.githubusercontent.com/62121649/189704982-baecdb00-c8a5-44ea-af21-31933e7dc972.png">

👍

**Reply by sunglee** [reply]
@mfix-stripe Thanks for the response! One more question. Do I need to install `@markdoc/next.js` even if I am not using Next.js?

**Reply by mfix-stripe** [reply]
Nope! You only need to install it if you are using it in your `next.config.js`.
